### DISCUSSION TITLE
When submitting an empty value, the result differs depending on the presence of ref

### DISCUSSION THREAD

**Discussion by adwd** [react-hook-form/react-hook-form#5151] [Q&A] [answered]
**Describe the bug**
If you submit without touching input, the value submitted will be different when using ref and when not using ref.

**To Reproduce**

1. Create a simple form without providing `defaultValues` to `useForm`
2. Create a same form of 1. but without using `ref`
3. submit form without touching input
4. If you use ref, empty characters will be submitted; if you don't use ref, undefined will be submitted.

**Codesandbox link (Required)**
https://codesandbox.io/s/react-hook-form-1qcoc

**Expected behavior**
Empty characters will be submitted even if you are not using ref.

**Screenshots**
<img width="543" alt="スクリーンショット 2021-05-11 20 54 58" src="https://user-images.githubusercontent.com/7473222/117811171-32fc4280-b29b-11eb-8e45-b096906acbb6.png">

**Desktop (please complete the following information):**

- OS: macOS
- Browser: chrome
- Version: 90

**Comment by bluebill1049** [comment, marked as answer]
expected behavior, and here is the explanation:

- with `ref`, so the value is coming from native input, which defaults to an empty string
- without `ref`, there is no input associated and hence value is `undefined`

**Reply by bluebill1049** [reply]
let me know if the above make sense, happy to elaborate on it.

**Reply by adwd** [reply]
I got it. Thank you for your quick reply!

**Reply by bluebill1049** [reply]
no worries ❤️

**Reply by adwd** [reply]
@bluebill1049 I noticed this will break type 🤔 

<img width="1212" alt="スクリーンショット 2021-05-13 12 51 33" src="https://user-images.githubusercontent.com/7473222/118075050-39e59b00-b3ea-11eb-879d-5ed53cb83944.png">

**Reply by bluebill1049** [reply]
Hmmmm because it's a run time thing, hook form can't detect type, maybe its worth to run setValue if u want type safe.
